src/scheduler/work.rs

```diff
@@ -40,6 +45,22 @@ pub trait GCWork<VM: VMBinding>: 'static + Send {
     fn get_type_name(&self) -> &'static str {
         std::any::type_name::<Self>()
     }
+
+    /// Get a human-readable size of the work packet if it makes sense.
+    ///
+    /// This method is useful for debug purposes, and is especially useful when visualising the
+    /// work packets during a GC.
+    ///
+    /// The semantics is unspecified.  For work packets that contains a list of work items, such as
+    /// object references or edges, the size should usually be the number of items.  If size does
+    /// not make sense for the work packet, this method may return `None`.
+    ///
+    /// This method will be called before the execution of every work packet, so it must be fast.
+    /// If it is impractical to report the exact size, this method can return approximate size, or
+    /// return `None`.
+    fn debug_get_size(&self) -> Option<usize> {
```

Comment:
One thing to note: this size may not always correlate with the time length of a packet's execution. This is the initial size of a work packet. Packets like `ProcessEdgesWork` may generate new work, and do it in the same work packet. That is not reflected by this number.

Reply:
My understanding is that ProcessEdgesWork might generate more packets, and store the packets in a worker's local queue that will be executed without pushing to the global queue. But they are still distinct packets.

Perhaps @wenyuzhao can clarify.

Reply:
I think one problem with the current tracepoints (which is not related to Kunshan's PR) is that packets pushed to the local queue, like object scanning in MMTk/OpenJDK, doesn't use the normal worker loop, so the work tracepoints might not fire (if you Zoom in on a traced obtained from OpenJDK, you can see gaps between ProcessEdges that are object scanning pakcets)

Reply:
`ScanObjectsWork` generated from `ProcessEdgesWork` is done in the same `ProcessEdgesWork` by default. This is hidden from our profiling code (work packet stats, and bpf tracing point). But new `ProcessEdgesWork` packets from `ScanObjectsWork` are pushed to the bucket, and will be shown correctly.

Reply:
I am planning to do an independent PR to probe the time of sub-tasks.  Therefore, `ProcessEdgesWork` will have two sub-tasks: processing edges and scanning objects.  The time of each of them will be presented separately.